src/applications/MachineShopSimulator.java

```diff
@@ -26,50 +26,40 @@
      */
     static boolean moveToNextMachine(Job theJob, SimulationResults simulationResults) {
         if (theJob.getTaskQ().isEmpty()) {// no next task
-            simulationResults.setJobCompletionData(theJob.getId(), timeNow, timeNow - theJob.getLength());
+            simulationResults.setJobCompletionData(theJob.getId(), getTimeNow(), getTimeNow() - theJob.getLength());
             return false;
         } else {// theJob has a next task
                 // get machine for next task
             int p = ((Task) theJob.getTaskQ().getFrontElement()).getMachine();
             // put on machine p's wait queue
             machine[p].getJobQ().put(theJob);
-            theJob.setArrivalTime(timeNow);
+            theJob.setArrivalTime(getTimeNow());
             // if p idle, schedule immediately
-            if (eList.nextEventTime(p) == largeTime) {// machine is idle
+            if (eList.nextEventTime(p) == getLargeTime()) {// machine is idle
                 changeState(p);
             }
             return true;
         }
     }
 
     /**
-     * change the state of theMachine
+     * change the state of theMachineJobCom
      * 
      * @return last job run on this machine
      */
     static Job changeState(int theMachine) {// Task on theMachine has finished,
                                             // schedule next one.
-        Job lastJob;
+        Job lastJob = null;
         if (machine[theMachine].getActiveJob() == null) {// in idle or change-over
                                                     // state
-            lastJob = null;
-            // wait over, ready for new job
-            if (machine[theMachine].getJobQ().isEmpty()) // no waiting job
-                eList.setFinishTime(theMachine, largeTime);
-            else {// take job off the queue and work on it
-                machine[theMachine].setActiveJob((Job) machine[theMachine].getJobQ()
-                        .remove());
-                machine[theMachine].setTotalWait(machine[theMachine].getTotalWait() + timeNow
-                        - machine[theMachine].getActiveJob().getArrivalTime());
-                machine[theMachine].setNumTasks(machine[theMachine].getNumTasks() + 1);
-                int t = machine[theMachine].getActiveJob().removeNextTask();
-                eList.setFinishTime(theMachine, timeNow + t);
-            }

```

Comment:
I really like how you guys refactored this logic of changeState() out. It really cleans of the readability of the method. :+1: